filters/auth/oauth.go

```diff
@@ -280,6 +314,13 @@ func (s *tokeninfoSpec) CreateFilter(args []interface{}) (filters.Filter, error)
 	return f, nil
 }
 
+// Close cleans-up the quit channel used for this spec
+func (s *tokeninfoSpec) Close() {
+	if s.authClient.quit != nil {
```

Comment:
A spec can create filters.
I think you have to close it from the `filter`, that's why you get a nil ptr panic.
See https://github.com/zalando/skipper/pull/707/files#diff-82513d1c973748c3c6b7fa0677fd3fd4L259, which shows that you store it in the filter and not in the spec.